src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs

```diff
@@ -166,6 +168,11 @@ string script
             Expression<Func<TWorkflow, TerminateTaskInput>> input
         ) => AddAndReturnBuilder(new TerminateTaskBuilder(reference.Body, input.Body));
 
+        public void SetOutput<F>(Expression<Func<TWorkflow, F>> input) where F : WorkflowOutput
```

Comment:
Shouldn't this method be parameterized with the type already specified as wf output?

Reply:
There is no point in specifying workflow output then.

Reply:
TWorkflow is ITypedWorkflow here and there is no compile time relationship with the input and output parameters. But, I will move this PR back to draft and try to think of some better way.

Reply:
You are right, I did not notice that.

Reply:
I think it makes sense to introduce a constraint for TWorkflow to be of type Workflow<TInput, TOutput>. Technically that would break SemVer rules but in this particular case it is highly unlikely anyone will be affected.

Reply:
I kind of implemented this, but I am not really happy with how it works. Take a look

Reply:
This is ok, I think there is no other way of implementing this. I guess at some point we will drop this <TWorkflow> builder and use one with specific types but we will see.